### PR TITLE
Improved DSU Module

### PR DESCRIPTION
**Pull request by SansPapyrus683**
Added Python support and made the solutions consistent with the initial classes provided.

### CODE REVIEW

File: content/4_Gold/DSU.mdx [hunk 1]
```diff
@@ -86,106 +85,146 @@ $x$ otherwise.
...
+	// returns whether the merge changed connectivity
+	public boolean union(int x, int y) {
+		int xRoot = find(x);
+		int yRoot = find(y);
+		if (xRoot == yRoot) {
+			return false;
+		}
+		if (sizes[xRoot] < sizes[yRoot]) {
+			parents[xRoot] = yRoot;
+			sizes[yRoot] = sizes[xRoot] + sizes[yRoot];
```
Comment (maggie-j-liu) [inline content/4_Gold/DSU.mdx:141]: can be
```suggestion
			sizes[yRoot] += sizes[xRoot];
```
but up to you

File: content/4_Gold/DSU.mdx [hunk 2]
```diff
@@ -86,106 +85,146 @@ $x$ otherwise.
 </Resources>
 
 ```cpp
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (dongliu0426) [inline content/4_Gold/DSU.mdx:89]: no need to include this; its just a dsu template

File: content/4_Gold/DSU.mdx [hunk 3]
```diff
@@ -86,106 +85,146 @@ $x$ otherwise.
...
 }
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+class DisjointSets:
+	def __init__(self, size: int) -> None:
+		self.parents = [-1 for _ in range(size)]
```
Comment (maggie-j-liu) [inline content/4_Gold/DSU.mdx:157]: also up to you
```suggestion
		self.parents = [-1] * size
```

File: content/4_Gold/DSU.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply content/4_Gold/DSU.mdx:89]: i feel like all code snippets should be standalone runnable, but if you really want to

File: content/4_Gold/DSU.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply content/4_Gold/DSU.mdx:157]: not really necessary tbh, and not like it helps readability

File: content/4_Gold/DSU.mdx [hunk 4]
```diff
@@ -86,106 +85,146 @@ $x$ otherwise.
...
-	bool unite(int x, int y) { // union by size
-		x = get(x), y = get(y); if (x == y) return 0;
-		if (e[x] > e[y]) swap(x,y);
-		e[x] += e[y]; e[y] = x; return 1;
+	
+	bool unite(int x, int y) {  // union by size
+		x = get(x), y = get(y);
+		if (x == y) return false;
+		if (e[x] > e[y]) swap(x, y);
+		e[x] += e[y]; e[y] = x; return true;
```
Comment (dolphingarlic) [inline content/4_Gold/DSU.mdx:106]:
```suggestion
		e[x] += e[y]; e[y] = x;
		return true;
```

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
Idk Java and the Python code looks fine so this is all I have to say

File: content/4_Gold/DSU.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply content/4_Gold/DSU.mdx]: also change lines 145, 264 and 267

File: content/4_Gold/DSU.mdx [hunk 5]
```diff
@@ -197,49 +237,56 @@ import java.io.*;
...
+		// returns whether the merge changed connectivity
+		public boolean union(int x, int y) {
+			int xRoot = find(x);
+			int yRoot = find(y);
+			if (xRoot == yRoot) {
+				return false;
+			}
+			if (sizes[xRoot] < sizes[yRoot]) {
+				parents[xRoot] = yRoot;
+				sizes[yRoot] += sizes[yRoot];
```
Comment (maggie-j-liu) [inline content/4_Gold/DSU.mdx:264]:
```suggestion
				sizes[yRoot] += sizes[xRoot];
```
typo?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: content/4_Gold/DSU.mdx (hunk 2 above)
Comment (dolphingarlic) [reply content/4_Gold/DSU.mdx:89]: I think this doesn't really matter either way so I'll just close and merge

**Pull request merged**